utils/qfirehose/Makefile

```diff
@@ -0,0 +1,57 @@
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=qfirehose
+PKG_VERSION:=1.4.9
+PKG_RELEASE:=1
+
+PKG_SOURCE:=$(PKG_VERSION).tar.gz
+PKG_SOURCE_URL:=https://github.com/nippynetworks/qfirehose/archive/refs/tags/
+PKG_HASH:=b7c04f9356823c6ee0f4ca152e8fd2015f34b95490cea68461a060993befadef
+
+PKG_MAINTAINER:=Oskari Rauta <oskari.rauta@gmail.com>
+PKG_LICENSE:=
+PKG_LICENSE_FILES:=NOTICE
```

Comment:
> https://github.com/nippynetworks/qfirehose/blob/f77d6729dcf72295e2c024782946cf8ac8a2d8c1/sahara_protocol.c#L13
> **Quectel Wireless Solution Proprietary and Confidential.**

I'm not sure if it's good to add it here.

Reply:
No, it is not. Only licenses from [this list](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/licensing-a-repository) are allowed if I remember correctly. And it is also in conflict with what NOTICE file says. I would keep it as is.

Reply:
> No, it is not. Only licenses from [this list](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/licensing-a-repository) are allowed if I remember correctly.

Nah, I meant add this package to the repo.

> And it is also in conflict with what NOTICE file says. I would keep it as is.

Yes, it's conflict with the NOTICE, but all source code (*.c & *.h) includes this claim.

And even in NOTICE file,
> **For those who is not Quectel's customer, all rights are closed, and any copying and commercial development over this progrma is not allowed.**

How to define "customer"?

Reply:
> Nah, I meant add this package to the repo.

Sorry, I am not sure what you mean by this- add this license to spdx?

> > **For those who is not Quectel's customer, all rights are closed, and any copying and commercial development over this progrma is not allowed.**
> 
> How to define "customer"?

Simple. Those who find this useful, meaning: owners of quectel's device. It is unlikely that this would even work for other brands, but in case it would- I wouldn't dare to try, atleast if I have any value for that other brand device, due to very high bricking risk.

Who then could abuse this? Other manufacturers who might want to use quectel's code to provide a method for firmware update on their devices, although, I am pretty sure that all we can be counted outside of that fear; if I'd be one of them, I would go directly to the source and it would have nothing to do with openwrt package. Actually, even other manufacturers are free to do whatever they want, as long as they own even one quectel product capable to use firehose updates.

Though, I am not judge in any court and not expert, so this is just IMHO...